src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -64,50 +64,29 @@ internal class AssemblyLoader
                     pathNameCriteria,
                     reflectionCriteria,
                     logger);
-
-            int count = 0;
+            
+            var loadedAssemblies = new List<Assembly>();
             List<string> discoveredAssemblyLocations = loader.DiscoverAssemblies();
             foreach (var pathName in discoveredAssemblyLocations)
             {
                 loader.logger.Info("Loading assembly {0}...", pathName);
+
                 // It is okay to use LoadFrom here because we are loading application assemblies deployed to the specific directory.
                 // Such application assemblies should not be deployed somewhere else, e.g. GAC, so this is safe.
-                Assembly.LoadFrom(pathName);
-                ++count;
-            }
-            loader.logger.Info("{0} assemblies loaded.", count);
-            return discoveredAssemblyLocations;
-        }
-
-        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger, IServiceProvider serviceProvider) where T : class
-        {
-            try
-            {
-                var assembly = Assembly.Load(new AssemblyName(assemblyName));
-                var foundType =
-                    TypeUtils.GetTypes(
-                        assembly,
-                        type => typeof(T).IsAssignableFrom(type) && !type.GetTypeInfo().IsInterface,
-                        logger).FirstOrDefault();
-                if (foundType == null)
+                try
                 {
-                    return null;
+                    loadedAssemblies.Add(Assembly.LoadFrom(pathName));
```

Comment:
hmmm, I'm a little scared of still using `LoadFrom` now that we are getting rid of the handling of AssemblyResolve (are we?). Not sure if this can go away. At least I'm glad that you are returning the assembly itself instead of the string name.
But didn't follow the entire approach, it's merely a heads up. I can take a closer look tomorrow at the office

Reply:
This should be correct - it's used only when adding assembly parts. We still have the assembly resolve hook during codegen (this occurs after assembly resolution is intercepted)

Reply:
Or... do we need to try a direct `Load` instead?